content/en/agent/troubleshooting/hostname_containers.md

```diff
@@ -0,0 +1,240 @@
+---
+title: Hostname Detection in Containers
+kind: documentation
+---
+
+Many features in Datadog rely on the Agent to provide an accurate hostname for monitored hosts. While this is straightforward when the Agent runs directly on a host, the hostname resolution process is different when the Agent runs in a containerized environment.
+
+Since version **7.40**, the Agent properly recognizes failed hostname resolution in containerized environments. Without a resolved hostname, the Agent exits with an error shortly after it starts.
+
+When that happens, the following `ERROR` message is printed in the logs:
+```
+Error while getting hostname, exiting: unable to reliably determine the host name. You can define one in the agent config file or in your hosts file
+```
+
+Encountering this error usually means that some part of the Agent configuration is incorrect. Use the following information to resolve various common cases of this misconfiguration.
+
+## Kubernetes hostname errors
+
+On Kubernetes, a hostname error usually means the Agent cannot access at least one of:
+* Kubelet API
+* Cloud provider metadata endpoint
+* Container runtime API
+
+Some Kubernetes distributions require a dedicated configuration, so verify that your setup is aligned with our [recommended Kubernetes setup][1].
+
+### Accessing the Kubelet API 
+
+Make sure the Agent can access the Kubelet API. When it can, the Agent prints this log:
+```
+Successful configuration found for Kubelet, using URL: ******
+```
+
+The Kubernetes RBAC permissions are set automatically by our official [Helm chart][2], the [Datadog Operator][3] and our official [manifests][4]. If you use a different solution to deploy the Agent, make sure the following permissions are present in a `Role` or `ClusterRole` that is bounded to the Agent service account:
+
+```yaml
+rules:
+  - apiGroups: # Kubelet connectivity
+      - ""
+    resources:
+      - nodes/metrics
+      - nodes/spec
+      - nodes/proxy
+      - nodes/stats
+    verbs:
+      - get
+```
+
+The most common error that prevents connection to the Kubelet API is the verification of Kubelet TLS certificate. In many Kubernetes distributions the Kubelet certificate is either:
+* Not signed by the cluster CA.
+* Does not contain a SAN corresponding to the address it's reachable at.
+
+This prevents the Agent from connecting to the Kubelet API through HTTPS, because TLS verification is enabled by default.
+
+You can disable TLS verification by using dedicated parameters or by setting the `DD_KUBELET_TLS_VERIFY` variable for **all containers** in the Agent manifest:
```

Comment:
Is disabling TLS verification really something we recommend doing?

Reply:
In some case yes, when the certificat is not valid or self signed.

Reply:
Unfortunately when it does not work OOTB, it's _usually_ impossible to fix without disabling TLS verify.